modules/tensor_mechanics/include/materials/ADLAROMCreepStressUpdate.h

```diff
@@ -0,0 +1,215 @@
+  virtual ADReal computeResidual(const ADReal & effective_trial_stress,
+                                 const ADReal & scalar) override;
+  virtual ADReal computeDerivative(const ADReal & /*effective_trial_stress*/,
+                                   const ADReal & /*scalar*/) override
+  {
+    return _derivative;
+  }
+  virtual void computeStressFinalize(const ADRankTwoTensor & plastic_strain_increment) override;
+
+  /**
+   * Compute the limiting value of the time step for this material
+   * @return Limiting time step
+   */
+  virtual Real computeTimeStepLimit() override;
+
+  void computeROMStrainRate(const Real dt,
+                            const Real & mobile_dislocations_old,
+                            const Real & immobile_dislocations_old,
+                            const ADReal & trial_stress,
+                            const Real & effective_strain_old,
+                            const ADReal & temperature,
+                            const ADReal & environmental,
+                            ADReal & mobile_dislocation_increment,
+                            ADReal & immobile_dislocation_increment,
+                            ADReal & rom_effective_strain,
+                            ADReal & rom_effective_strain_derivative);
+
+  /// Function to check input values against applicability windows set by ROM data set
+  void checkInputWindows(std::vector<ADReal> & input);
+
+  /* Convert the input variables into the form expected by the ROM Legendre polynomials
+   * to have a normalized space from [-1, 1] so that every variable has equal weight
+   */
+  void convertInput(const std::vector<ADReal> & input,
+                    std::vector<std::vector<ADReal>> & converted,
+                    std::vector<std::vector<ADReal>> & dconverted);
+
+  /**
+   * Assemble the array of Legendre polynomials to be multiplied by the ROM
+   * coefficients
+   */
+  void buildPolynomials(const std::vector<std::vector<ADReal>> & rom_inputs,
+                        const std::vector<std::vector<ADReal>> & drom_inputs,
+                        std::vector<std::vector<std::vector<ADReal>>> & polynomial_inputs,
+                        std::vector<std::vector<std::vector<ADReal>>> & dpolynomial_inputs);
+
+  /**
+   * Arranges the calculated Legendre polynomials into the order expected by the
+   * ROM coefficients and ultiplies the Legendre polynomials by the ROM coefficients to compute the
+   * the predicted output values
+   */
+  void computeValues(const std::vector<std::vector<Real>> & coefs,
+                     const std::vector<std::vector<std::vector<ADReal>>> & polynomial_inputs,
+                     const std::vector<std::vector<std::vector<ADReal>>> & dpolynomial_inputs,
+                     std::vector<ADReal> & rom_outputs,
+                     std::vector<ADReal> & drom_outputs);
+
+  /* Computes the output variable increments from the ROM predictions by bringing out of the
```

Comment:
Doxygen might require the first comment line to be 

```
/**
```

Reply:
whoops